dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + submitterGroup.getID() + "/object")
+            ).andExpect(status().isOk());
+        } finally {
+            if (collection != null) {
+                CollectionBuilder.deleteCollection(collection.getID());
+            }
+            if (community != null) {
+                CommunityBuilder.deleteCommunity(community.getID());
+            }
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+            if (worfklowGroup != null) {
+                GroupBuilder.deleteGroup(worfklowGroup.getID());
+            }
+            if (submitterGroup != null) {
+                GroupBuilder.deleteGroup(submitterGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void getGroupObjectNotFoundTest() throws Exception {
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+        Group adminGroup = null;
+        try {
+            context.turnOffAuthorisationSystem();
+            adminGroup = groupService.create(context);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
+
+            String authToken = getAuthToken(admin.getEmail(), password);
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + adminGroup.getID() + "/object")
+            ).andExpect(status().isNotFound());
+        } finally {
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void getGroupObjectUnauthorizedTest() throws Exception {
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+        Group adminGroup = null;
+        try {
+            context.turnOffAuthorisationSystem();
+            adminGroup = groupService.create(context);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
+
+            getClient().perform(
+                    get("/api/eperson/groups/" + adminGroup.getID() + "/object")
+            ).andExpect(status().isUnauthorized());
+        } finally {
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void patchNameTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        String testGroupName = "Test group";
+        Group group = GroupBuilder.createGroup(context)
+                .withName(testGroupName)
+                .build();
+
+
+        String newName = "New test name";
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/name", newName);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
```

Comment:
This test looks great. It's not hardcoding JSON & it's using the GroupBuilder to create a test Group.  The other tests above should use this same pattern to make them easier to read/review.

**However, the only thing missing here is a `context.restoreAuthSystemState();`**